src/test/java/study/step3/LottoNumberTest.java

```diff
@@ -0,0 +1,27 @@
+package study.step3;
+
+import static org.assertj.core.api.AssertionsForClassTypes.*;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import study.step3.domain.LottoNumber;
+
+public class LottoNumberTest {
+
+    @Test
+    @DisplayName("유효성 테스트")
+    void valid() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> LottoNumber.of(50));
+        Assertions.assertThrows(IllegalArgumentException.class, () -> LottoNumber.of(-1));
+    }
+
+    @ParameterizedTest(name = "로또 번호 생성 테스트")
+    @ValueSource(strings = {"1", "45"})
+    public void of(String input) {
+        assertThat(LottoNumber.of(Integer.parseInt(input)).toString()).isEqualTo(input);
+    }
```

Comment:
`given-when-then 패턴`에 맞춰 테스트를 작성하면 동료 개발자가 코드를 파악하는데 훨씬 좋습니다 😄 
[Given-When-Then Pattern](https://brunch.co.kr/@springboot/292)

추가로 테스트 하실때 클래스에 `equals(), hashCode()` 메소드 재정의에 대해서 고민해보시면 좋을것 같아요.

TDD로 객체지향 프로그래밍을 한다고 했을때 객체에서 값을 꺼내서 비교하는 것이 아니라 
`객체 자체를 비교` 비교하는 것을 말씀드릴려고 했습니다. 아마 포비님이 강의에서도 말씀하실꺼에요 😄

현재는 LottoNumber에 캐싱이 적용되어 있어 괜찮지만 
`equals(), hashCode()` 메소드를 재정의 하지 않으면 아래 테스트는 깨지게 될꺼에요.

```suggestion
    @ParameterizedTest
    @ValueSource(ints = {1, 45})
    @DisplayName("유효 범위 내의 로또 번호 생성 테스트")
    void valid_lotto_number_test(int number) {
        // given
        LottoNumber lottoNumber = LottoNumber.of(number);

        // when & then
        assertThat(lottoNumber).isEqualTo(LottoNumber.of(number));
    }
```

Reply:
넵 감사합니다 :)